Sources/SourceKitLSP/TestDiscovery.swift

```diff
@@ -497,7 +500,60 @@ fileprivate extension Array<AnnotatedTestItem> {
   }
 }
 
+fileprivate extension Array<TestItem> {
+  /// If multiple testItems share the same ID we add more context to make it unique.
+  /// Two tests can share the same ID when two swift testing tests accept
+  /// arguments of different types, i.e:
+  /// ```
+  /// @Test(arguments: [1,2,3]) func foo(_ x: Int) {}
+  /// @Test(arguments: ["a", "b", "c"]) func foo(_ x: String) {}
+  /// ```
+  ///
+  /// or when tests are in separate files but don't conflict because they are marked
+  /// private, i.e:
+  /// ```
+  /// File1.swift: @Test private func foo() {}
+  /// File2.swift: @Test private func foo() {}
+  /// ```
+  ///
+  /// If we encounter one of these cases, we need to deduplicate the ID
+  /// by appending `/filename:filename:lineNumber`.
+  func deduplicateIds() -> [TestItem] {
+    var idCounts: [String: Int] = [:]
+    var result: [TestItem] = []
+
+    for element in self where element.children.isEmpty {
+      idCounts[element.id, default: 0] += 1
```

Comment:
Do you think it makes sense to have an early exit here if there are no ambiguous IDs? Ie. keep track whether we ever incremented `idCounts` for any value to `2` and if not, just `return self` without creating a second array.